server/server_test.go

```diff
@@ -126,14 +129,14 @@ func TestDefaultAddresses(t *testing.T) {
 		w.WriteHeader(204)
 	})
 
-	go server.Run()
+	go func() { require.NoError(t, server.Run()) }()
```

Comment:
I don't think require.NoError will work in a background goroutine.
https://pkg.go.dev/testing#T.FailNow

(there are multiple instances of this in the PR)